File: app/src/main/java/com/amary/app/data/moviecat/adapter/MovieCatAdapter.java
package com.amary.app.data.moviecat.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.amary.app.data.moviecat.R;
import com.amary.app.data.moviecat.model.MovieCatData;

import java.util.ArrayList;

public class MovieCatAdapter extends BaseAdapter {
    private Context context;
    private ArrayList<MovieCatData> movieCatData;

    public void setMovieCatData(ArrayList<MovieCatData> movieCatData) {
        this.movieCatData = movieCatData;
    }

    public MovieCatAdapter(Context context) {
        this.context = context;
        movieCatData = new ArrayList<>();
    }

    @Override
    public int getCount() {
        return movieCatData.size();
    }

    @Override
    public Object getItem(int position) {
        return movieCatData.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View view, ViewGroup viewGroup) {
        if (view ==null){
            view = LayoutInflater.from(context).inflate(R.layout.item_moviecat, viewGroup, false);
        }

        ViewHolder viewHolder = new ViewHolder(view);
        MovieCatData movieCatData = (MovieCatData) getItem(position);
        viewHolder.bind(movieCatData);
        return view;
    }

    private class ViewHolder{
        private ImageView imgPoster;
        private TextView txtJudul;
        private TextView txtTahunRilis;
        private TextView txtDurasi;
        private TextView txtGenre;

        ViewHolder(View view){
            imgPoster       = view.findViewById(R.id.img_poster);
            txtJudul        = view.findViewById(R.id.txt_judul);
            txtTahunRilis   = view.findViewById(R.id.txt_thn_rilis);
            txtDurasi       = view.findViewById(R.id.txt_durasi);
            txtGenre        = view.findViewById(R.id.txt_genre);
        }

        void bind(MovieCatData movieCatData){
            imgPoster.setImageResource(movieCatData.getPoster());
            txtJudul.setText(movieCatData.getJudul());
            txtTahunRilis.setText(movieCatData.getThnRilis());
            txtDurasi.setText(movieCatData.getDurasiMovie());
            txtGenre.setText(movieCatData.getGenreMovie());
        }
    }
}

File: app/src/main/java/com/amary/app/data/moviecat/adapter/MovieCatAdapter.kt
package com.amary.app.data.moviecat.adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.amary.app.data.moviecat.R
import com.amary.app.data.moviecat.activity.DetailMovieActivity
import com.amary.app.data.moviecat.model.MovieCatData
import kotlinx.android.synthetic.main.item_moviecat.view.*
import java.util.*
import kotlin.collections.ArrayList

class MovieCatAdapter(private val context: Context) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    private var movieCatData: MutableList<MovieCatData> = ArrayList()

    fun setMovieCatData(movieCatData: ArrayList<MovieCatData>) {
        this.movieCatData.addAll(movieCatData)
        notifyDataSetChanged()
    }

    private fun getItem(position: Int): MovieCatData {
        return movieCatData[position]
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_moviecat, parent, false)
        return ItemViewHolder(view)
    }

    override fun getItemCount(): Int {
        return movieCatData.size
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        (holder as ItemViewHolder).bind(getItem(position), context)
    }

    inner class ItemViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        fun bind(item: MovieCatData, context: Context) {
            itemView.img_poster.setImageResource(item.poster)
            itemView.txt_durasi.text = item.judul
            itemView.txt_thn_rilis.text = item.thnRilis
            itemView.txt_durasi.text = item.durasiMovie
            itemView.txt_genre.text = item.genreMovie

            itemView.setOnClickListener {
                val intent = Intent(context, DetailMovieActivity::class.java)
                intent.putExtra(DetailMovieActivity.EXTRA_MOVIE, item)
                context.startActivity(intent)
            }
        }

    }
}